### PR TITLE
Tree Widget: Zoom to element in stateless models tree

### PR DESCRIPTION
**Pull request by jasdom**
Closes https://github.com/iTwin/viewer-components-react/issues/867.

### CODE REVIEW

File: packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx [hunk 1]
```diff
@@ -45,8 +47,21 @@ export function TreeRenderer({
...
+            onNodeDoubleClick && event.detail !== 2 && onNodeClick?.(nodeId, isSelected, event);
+          }}
+          nodeProps={{
+            onDoubleClick: (event) => {
+              if (nodeProps.isDisabled || "type" in nodeProps.node || !isPresentationHierarchyNode(nodeProps.node)) {
+                return;
+              }
+              onNodeDoubleClick?.(nodeProps.node, !!nodeProps.isSelected);
+              // Select node to not lose highlight
+              !nodeProps.isSelected && onNodeClick?.(nodeProps.node, true, event);
```
Comment (grigasp) [inline packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx:61]: does it only loose selection in the tree or does this cause the node to be deselected and afterwards selected in unified selection?

File: packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx [hunk 2]
```diff
@@ -45,8 +47,21 @@ export function TreeRenderer({
       return (
         <TreeNodeRenderer
           {...nodeProps}
+          onNodeClick={(nodeId, isSelected, event) => {
```
Comment (saskliutas) [inline packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx:50]: Maybe we should handle double clicks in `onNodeClick` as now we have code related to double clicks in both events?

Comment (jasdom) [reply packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx]: Moved into `onNodeClick`.

File: packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx (hunk 1 above)
Comment (jasdom) [reply packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx]: The first click is handled as normal so it does deselect in unified selection if double clicked slowly enough. Seems to work the same as in the old models tree.

Comment (grigasp) [reply packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx]: As discussed, in "extended" selection mode a click on a selected node should not deselect it. I think that solves the problem and we won't need to select it back.

In other selection modes let's keep the behavior you have now.

File: packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeNodeRenderer.tsx [hunk 3]
```diff
@@ -19,11 +19,11 @@ type CoreTreeNodeRendererProps = ComponentPropsWithoutRef<typeof CoreTreeNodeRen
 type TreeNodeRendererProps = CoreTreeNodeRendererProps & TreeNodeRendererOwnProps;
 
 /** @internal */
-export function TreeNodeRenderer({ checkboxProps, ...props }: TreeNodeRendererProps) {
+export function TreeNodeRenderer({ checkboxProps, nodeProps, ...props }: TreeNodeRendererProps) {
   return (
     <CoreTreeNodeRenderer
       {...props}
-      nodeProps={{ className: cx("tw-tree-node", props.isSelected && "selected", props.className) }}
+      nodeProps={{ ...nodeProps, className: cx("tw-tree-node", props.isSelected && "selected", props.className) }}
```
Comment (saskliutas) [inline packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeNodeRenderer.tsx:26]: I guess `...nodeProps` can be removed? Or if you expect that we will need to pass custom nodeProps in the future same need to be done with `contentProps`

**Review by saskliutas — CHANGES_REQUESTED** [review]

File: packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeNodeRenderer.tsx (hunk 3 above)
Comment (jasdom) [reply packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeNodeRenderer.tsx]: Removed, we can add it when we need it.

File: packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx (hunk 1 above)
Comment (jasdom) [reply packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx]: Done.

Comment (grigasp) [reply packages/itwin/tree-widget/src/components/trees/stateless/common/components/TreeRenderer.tsx]: For tracking - the change was made in https://github.com/iTwin/presentation/pull/627 and consumed into tree widget with https://github.com/iTwin/viewer-components-react/pull/898.

**Review by saskliutas — APPROVED** [review]

**Pull request merged**